src/command_modules/azure-cli-profile/azure/cli/command_modules/profile/custom.py

```diff
@@ -42,6 +42,24 @@ def show_subscription(subscription=None, expanded_view=None):
         return profile.get_expanded_subscription_info(subscription)
 
 
+def get_access_token(subscription=None, resource=None):
+    '''
+    get AAD token to access to a specified resource
+    :param resource: Azure resource endpoints. Default to Azure Resource Manager
+    Use 'az cloud show' command for other Azure resources
+    '''
+    profile = Profile()
+    creds, subscription, tenant = profile.get_raw_token(subscription_id=subscription,
+                                                        resource=resource)
+    return {
+        'tokenType': creds[0],
+        'accessToken': creds[1],
+        'expiresOn': creds[2]['expiresOn'],
+        'subscription': subscription,
```

Comment:
Is the subscription name useful as well? Looks like this would be the id only.
I'd consider adding the name if it's easy enough to get.